### PR TITLE
ddl: wrap some functions for ddl job

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/34919

Problem Summary:

there are some functions to operate ddl jobs like `GetHistoryDDLJob` which is a method of meta, it is not easy to change the implementation because every caller should build a meta for it. So, wrapping them in another will be better.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: store/gcworker/gc_worker.go [hunk 1]
```diff
@@ -1881,12 +1881,9 @@ func (w *GCWorker) doGCPlacementRules(safePoint uint64, dr util.DelRangeTask, gc
 		}
 	})
 	if historyJob == nil {
-		err = kv.RunInNewTxn(context.Background(), w.store, false, func(ctx context.Context, txn kv.Transaction) error {
-			var err1 error
-			t := meta.NewMeta(txn)
-			historyJob, err1 = t.GetHistoryDDLJob(dr.JobID)
-			return err1
-		})
+		se := createSession(w.store)
```
Comment (tangenta) [inline store/gcworker/gc_worker.go:1884]: `createSession()` is much heavy. I think it may cause a performance regression.

File: executor/ddl.go [hunk 2]
```diff
@@ -608,7 +608,12 @@ func (e *DDLExec) executeRecoverTable(s *ast.RecoverTableStmt) error {
 }
 
 func (e *DDLExec) getRecoverTableByJobID(s *ast.RecoverTableStmt, t *meta.Meta, dom *domain.Domain) (*model.Job, *model.TableInfo, error) {
```
Comment (zimulala) [inline executor/ddl.go:610]: So could we remote the argument of `t`?

File: ddl/ddl.go [hunk 3]
```diff
@@ -976,9 +977,13 @@ func getDDLJobsInQueue(t *meta.Meta, jobListKey meta.JobListKeyType) ([]*model.J
 	return jobs, nil
 }
 
-// GetDDLJobs get all DDL jobs and sorts jobs by job.ID.
-func GetDDLJobs(txn kv.Transaction) ([]*model.Job, error) {
-	t := meta.NewMeta(txn)
+// GetAllDDLJobs get all DDL jobs and sorts jobs by job.ID.
+func GetAllDDLJobs(t *meta.Meta) ([]*model.Job, error) {
```
Comment (zimulala) [inline ddl/ddl.go:981]: Why are we doing this? I don't feel the need

Comment (xiongjiwei) [reply ddl/ddl.go:981]: well, it is like an entrance of "get all ddl jobs", it now gets from `meta`, in the future, we may add another way to get them and use a switch to control it.

Comment (zimulala) [reply ddl/ddl.go:981]: I think this can be changed in the future, there's no need to add the function of `getDDLJobs`.

Comment (xiongjiwei) [reply ddl/ddl.go:981]: Done, PTAL

**Comment by xiongjiwei** [comment]
@zimulala PTAL

File: ddl/ddl.go [hunk 4]
```diff
@@ -1057,3 +1057,28 @@ func IterAllDDLJobs(txn kv.Transaction, finishFn func([]*model.Job) (bool, error
 	}
 	return IterHistoryDDLJobs(txn, finishFn)
 }
+
+// GetAllHistoryDDLJobs get all the done ddl jobs.
```
Comment (zimulala) [inline ddl/ddl.go:1061]:
```suggestion
// GetAllHistoryDDLJobs get all the done DDL jobs.
```

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**